src/main/java/seedu/address/model/ContactTagIntegrationManager.java

```diff
@@ -42,6 +41,13 @@ public TagTree getTagTree() {
         return tagTree;
     }
 
+    /**
+     * Returns true if either the addressBook has {@code tag} or the tagTree has {@code tag}.
+     */
+    public boolean hasTag(Tag tag) {
+        return addressBook.hasTag(tag) || tagTree.hasTag(tag);
```

Comment:
does this imply that some tags (i assume highest parent level ones) are in `addressBook` but not `tagTree`?

Reply:
yep. the requirement for a `tag` to exist is that it must have at least one contact with the `tag` or the `tag` has at least one child-tag.

Reply:
`tagtree` handles `tag` to `tag` relations. `addressbook` handles `tag` to `person` relations. As such, one can be in one without being in the other as the requirement to exist in either is that it has at least one object assigned to it.